relay/cache/cached_accessor.go

```diff
@@ -1,24 +1,28 @@
 package cache
 
 import (
+	"context"
 	lru "github.com/hashicorp/golang-lru/v2"
+	"golang.org/x/sync/semaphore"
 	"sync"
 )
 
 // CachedAccessor is an interface for accessing a resource that is cached. It assumes that cache misses
 // are expensive, and prevents multiple concurrent cache misses for the same key.
 type CachedAccessor[K comparable, V any] interface {
 	// Get returns the value for the given key. If the value is not in the cache, it will be fetched using the Accessor.
-	Get(key K) (V, error)
+	// If the context is cancelled, the function may abort early. If multiple goroutines request the same key,
+	// cancellation of one request will not affect the others.
```

Comment:
nit: the cancellation of the first request can affect others (all of them get error)

Reply:
The way I implemented this class, cancellation of the first request should not cause there to be an error for the following requests. Do you see a bug in the implementation?

Reply:
Ok, I think it's about how to read "not affect", the failure of fetch is not the failure of first request, then this comment is correct. Anyway, it's fine to keep this.